globus_cli/commands/transfer.py

```diff
@@ -398,6 +421,16 @@ def process_batch_line(dest_path, source_path, recursive, external_checksum):
             recursive=recursive,
         )
 
+    for item in transfer_data["DATA"]:
+        if item["recursive"]:
+            has_recursive_items = True
+            break
+    else:
+        has_recursive_items = False
```

Comment:
I think this is easier to read as
```python
has_recursive_items = any(item["recursive"] for item in transfer_data["DATA"])
```
as part of my general thinking that `for-else` can be confusing.

But it's not a big deal -- definitely not enough to hold up this PR.

Reply:
Oh! I do agree that this is cleaner, I'll try to use that pattern in the future at the very least if we don't want to go back and use it here